core/common/src/main/java/alluxio/extensions/ExtensionFactoryRegistry.java

```diff
@@ -131,19 +131,18 @@ private synchronized void init() {
     Preconditions.checkArgument(path != null, "path may not be null");
 
     List<T> factories = new ArrayList<>(mFactories);
+    List<T> eligibleFactories = select(path, conf, factories);
```

Comment:
You mean the first time `findAll ` invoked, the `mFactories ` should be empty, and this method will run like original logic, later new `findAll ` will not load the extension or lib jar at all? So please show the effect.

Reply:
> You mean the first time `findAll ` invoked, the `mFactories ` should be empty, and this method will run like original logic, later new `findAll ` will not load the extension or lib jar at all? So please show the effect.

Yes, the new 'findAll' will not dynamicly load the ext or lib jar after the first load done. However, we can restart the alluxio master or worker to load the new  updated jar.

Reply:
Please highlight add this important limitation to the related issue or the description of this PR. It would be better if you write an approach description shortly to let reviewer know your solution and make it easy to review this PR